lib/ruby_lsp/listeners/definition.rb

```diff
@@ -182,10 +182,10 @@ def handle_instance_variable_definition(name)
         # If by any chance we haven't indexed the owner, then there's no way to find the right declaration
       end
 
-      sig { params(message: String, receiver_type: T.nilable(String), inherited_only: T::Boolean).void }
+      sig { params(message: String, receiver_type: T.nilable(TypeInferrer::Type), inherited_only: T::Boolean).void }
       def handle_method_definition(message, receiver_type, inherited_only: false)
         methods = if receiver_type
-          @index.resolve_method(message, receiver_type, inherited_only: inherited_only)
```

Comment:
If the `receiver_type` is a guessed type, should we also provide `untyped` candidates (the `else` condition), so users won't lose the definitions they should've had if the guessed type is wrong?